cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java

```diff
@@ -528,11 +528,14 @@ public static void populateRackInfoForReplicationFactorChange(Map<Short, Set<Str
     Map<String, Double> balancednessCostByGoal = new HashMap<>(goals.size());
     // Step-1: Get weights.
     double weightSum = 0.0;
+    double previousGoalPriorityWeight = 1;
     for (int i = goals.size() - 1; i >= 0; i--) {
       Goal goal = goals.get(i);
-      double cost = Math.pow(priorityWeight, (goals.size() - 1 - i)) * (goal.isHardGoal() ? strictnessWeight : 1);
+      double currentGoalPriorityWeight = priorityWeight * previousGoalPriorityWeight;
```

Comment:
Looks with this change, the behavior changes.
for the last goal(first iteration), its weight should be 1.0(original it is `pow(priorityWeight, 0)`, but now it is `priorityWeight`

Reply:
Good catch!